203 view liked posts & 198 view liked posts on profile

jerohero opened the pull request:
closes #203
closed #198

 **What is it supposed to do**
Everyone can see who liked a post and can see what posts a user liked on their profile (yes I put two different use cases in one branch)

 **How can we test your branch**
- Go to a post and click on the amount of a like to open the list (shouldn't be possible if there are no likes)
- Click on a profile in the list, test whether it's sorting properly or whatever
- Go to a profile and look at the liked posts. See if this updates properly and is sorted.

 **What broweser have been checked**
- [x] Chrome
- [x] Firefox
- [ ] Opera
- [ ] Edge
- [ ] Safari

jerohero committed:
Initialized list

jerohero committed:
Merge branch '192-like-posts' into 198-view-liked-posts

jerohero committed:
Created popup

jerohero committed:
Likes are shown in frontend with some styling

jerohero committed:
Removed code that fills liked list with duplicate values

jerohero committed:
Fetch user's likes in backend

jerohero committed:
Shows user's likes on profile

jerohero committed:
Sort user's likes by descending date

jerohero committed:
Merge branch 'develop' into 198-view-liked-posts

# Conflicts:
#	frontend/src/pages/profile/index.js

jerohero committed:
Can only open likes list if there are 1 or more likes

jerohero committed:
User's (placeholder) title is shown + likes are sorted

jerohero committed:
Profiles in likes list are Link elements instead of history push

jerohero committed:
OP can't like their own posts + small refactoring before merge

AjobK reviewed (changes requested):
Change to fontawesome icons, see screenshot for style reference too:
![image](https://user-images.githubusercontent.com/35965716/112724774-37e87b00-8f15-11eb-8785-b676a2a09bbe.png)

Rename feathers to likes

Make overlay fixed, not absolute

AjobK reviewed (changes requested):
![image](https://user-images.githubusercontent.com/35965716/112727835-9ec16080-8f24-11eb-8d52-87a942a95136.png)

When hallway testing it appeared that spamming the like button results in multiple likes by a single person.

jerohero committed:
Merge branch 'develop' into 198-view-liked-posts

jerohero committed:
Implemented Ajob's feedback

+ Changed to FontAwesome icons
+ Implemented style reference
+ Renamed feathers to likes
+ Made overlay fixed, not absolute

AjobK reviewed (changes requested):
![image](https://user-images.githubusercontent.com/35965716/113441353-68855480-93ee-11eb-8b26-dd916fb79a59.png)
Unable to scroll in this window

Please make the profile_id and post_id combination unique in the post_like table, so there will never be a duplicate like (In other words, one person liking twice).

Also turn the 'X' on the right into an icon.
![image](https://user-images.githubusercontent.com/35965716/113441403-8ce13100-93ee-11eb-96f8-08344908abcd.png)
I suggest this one: https://fontawesome.com/icons/times-circle?style=solid

Change the view into the example previously shown in discord, except for the 'Feathers' part which should be 'Likes' and obviously the icons:
![image](https://user-images.githubusercontent.com/35965716/113441557-c6b23780-93ee-11eb-92f0-5a703ffb110c.png)

AjobK commented on backend/app/dao/postDao.ts:
```diff
@@ -56,6 +56,17 @@ class PostDAO {
 
+    public async getRecentUserLikesByProfileId(profileId: number, limit): Promise<any> {
+        if (!profileId)
+            return null
```
Newline under return

AjobK commented on backend/app/dao/postDao.ts:
```diff
@@ -56,6 +56,17 @@ class PostDAO {
+        if (!profileId)
+            return null
+        const repository = await DatabaseConnector.getRepository('PostLike')
+        const likes = await repository.find({ where: {profile: profileId}, relations: ['post', 'profile'], take: limit})
```
newline under declaration

AjobK commented on backend/app/dao/postDao.ts:
```diff
@@ -56,6 +56,17 @@ class PostDAO {
+            return null
+        const repository = await DatabaseConnector.getRepository('PostLike')
+        const likes = await repository.find({ where: {profile: profileId}, relations: ['post', 'profile'], take: limit})
+        if (!likes || likes.length <= 0) return
```
newline under

AjobK commented on backend/app/dao/postDao.ts:
```diff
@@ -56,6 +56,17 @@ class PostDAO {
+        const repository = await DatabaseConnector.getRepository('PostLike')
+        const likes = await repository.find({ where: {profile: profileId}, relations: ['post', 'profile'], take: limit})
+        if (!likes || likes.length <= 0) return
+        return likes.sort((a, b) => {
```
a and b is cryptic. Please use understandable variable names

AjobK commented on backend/app/service/postService.ts:
```diff
@@ -188,11 +183,27 @@ class PostService {
+
+            return res.status(200).json(likes)
+        }
+        else
```
Please use either shorthand or the extended version. Don't combine if () {} with a normal else without curly braces. In this case use curly braces for the else

AjobK commented on backend/app/service/postService.ts:
```diff
@@ -188,11 +183,27 @@ class PostService {
+        const foundLikes = await this.dao.getRecentUserLikesByProfileId(foundUser.id, 8)
+
+        if (req.params.username && foundLikes) {
+            const likes = []
```
newline under

AjobK commented on frontend/src/components/postLike/index.js:
```diff
@@ -3,19 +3,31 @@ import styles from './postLike.scss'
 import Axios from 'axios'
 import { withRouter } from 'react-router'
+import { PostLikesList } from '../../components'
+import Like from '../../static/icons/heart-solid.svg';
```
Please use icons differently. Use navigation for reference. There is a seperate component for icons

AjobK replied:
Also remove semicolon

AjobK commented on frontend/src/components/postLike/index.js:
```diff
@@ -30,31 +42,50 @@ class PostLike extends Component {
-                    &#10084;
+                    <img src={Like} alt={'Like'}/>
                 </button>
+                {this.state.showLikes && (
```
Space before 'this.state.showLikes' to make it more readable and clean. Same for cases above such as '{likesAmount' which should be '{ likesAmount'

AjobK commented on frontend/src/components/postLikesList/index.js:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    loadLikes = () => {
+        const path = window.location.pathname.split('/').filter(i => i != '').pop()
```
Newline under

AjobK commented on frontend/src/components/postLikesList/index.js:
```diff
@@ -0,0 +1,78 @@
+        const path = window.location.pathname.split('/').filter(i => i != '').pop()
+        Axios.get(`${this.props.store.defaultData.backendUrl}/post/like/${path}`)
+            .then(res => {
+                const likes = []
```
Newline under

AjobK commented on frontend/src/components/postLikesList/index.js:
```diff
@@ -0,0 +1,78 @@
+                        level: 0,
+                        title: 'Software Engineer'
+                    }
+                    likes.push(userLike)
```
newline above

AjobK commented on frontend/src/components/postLikesList/index.js:
```diff
@@ -0,0 +1,78 @@
+                        <span className={`${styles.closeLikes}`} onClick={this.props.closeLikesList}>x</span>
+                    </header>
+                    <ul className={`${styles.likesList}`}>
+                        {this.state.likes.map((like, index) => {
```
space between { and this resulting in '{ this.state.likes.....'

AjobK commented on frontend/src/pages/profile/index.js:
```diff
@@ -51,6 +55,16 @@ class Profile extends App {
+        })
+        .catch(() => {
+          // this.setState({ error: true })
+        })
```
Remove comment

AjobK commented on frontend/src/components/postLikesList/index.js:
```diff
@@ -0,0 +1,78 @@
+import styles from './postLikesList.scss'
+import { inject, observer } from 'mobx-react'
+import Axios from "axios"
+import { Link } from "react-router-dom";
```
Please use single quotes and remove semicolon

AjobK commented on frontend/src/components/postLikesList/index.js:
```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react'
+import styles from './postLikesList.scss'
+import { inject, observer } from 'mobx-react'
+import Axios from "axios"
```
Single quotes

AjobK reviewed (changes requested):
A few other minor things

jerohero committed:
Fixed unique constraints for post likes and fixed scrolling issue

jerohero committed:
Implemented new requested changes

+ Image tags to Icon components
+ Styling
+ Newlines, quotes etc

AjobK committed:
Some changes, please do rest :-)

jerohero committed:
i add the thing =p

swag LOL

AjobK commented on backend/app/service/postService.ts:
```diff
@@ -188,11 +179,29 @@ class PostService {
+            })
+
+            return res.status(200).json(likes)
+        }
```
else on this line

AjobK commented on frontend/src/components/postBanner/index.js:
```diff
@@ -2,7 +2,7 @@ import React, { Component } from 'react'
 import styles from './postBanner.scss'
 import { inject, observer } from 'mobx-react'
 import { Icon } from '..';
```
semicolon

AjobK reviewed (changes requested):
minor changes and show like for owner of post

jerohero committed:
Show likes when user owns the post, without them being able to like

ryankroon00 committed:
merged with new develop

AjobK reviewed (approved):
Looks good! :-)
